evm_arithmetization/src/witness/transition.rs

```diff
@@ -382,30 +382,35 @@ where
             .map_err(|_| ProgramError::InvalidJumpDestination)?;
 
         if !self.generate_jumpdest_analysis(dst as usize) {
-            let gen_state = self.get_mut_generation_state();
-            let (jumpdest_bit, jumpdest_bit_log) = mem_read_gp_with_log_and_fill(
-                NUM_GP_CHANNELS - 1,
-                MemoryAddress::new(
-                    gen_state.registers.context,
-                    Segment::JumpdestBits,
-                    dst as usize,
-                ),
-                gen_state,
-                &mut row,
-            );
-
             row.mem_channels[1].value[0] = F::ONE;
 
-            if gen_state.registers.is_kernel {
-                // Don't actually do the read, just set the address, etc.
-                let channel = &mut row.mem_channels[NUM_GP_CHANNELS - 1];
-                channel.used = F::ZERO;
-                channel.value[0] = F::ONE;
-            } else {
-                if jumpdest_bit != U256::one() {
-                    return Err(ProgramError::InvalidJumpDestination);
+            // We skip jump destinations verification with `cdk_erigon`.
+            #[cfg(not(feature = "cdk_erigon"))]
```

Comment:
Since we don't do JDA, I think we must check manually that the dest address is indeed a `JUMPDEST`.

Reply:
What does it change? The type2 won't be performing any checks on this. A malicious prover could very well skip this altogether. Or did you mean just as safeguard for us? like an early error / panic?

Reply:
Ah no indeed we could still process honestly invalid JD, let me change this.

Reply:
Let me know if last commit works for you